### PR TITLE
Avoid loadMedia copy and do some cleanups

### PR DESCRIPTION
**Pull request by nerzhul**

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -690,7 +686,7 @@ bool Client::loadMedia(const std::string &data, const std::string &filename)
 
 		// Create an irrlicht memory file
 		io::IReadFile *rfile = irrfs->createMemoryReadFile(
-				*data_rw, data_rw.getSize(), "_tempreadfile");
+				(void*)data.c_str(), data.size(), "_tempreadfile");
```
Comment (est31) [inline src/client.cpp]: Best would be:

1. Adding an #ifdef for irrlicht 1.9 with the "direct" way that doesn't do a cast

1. in the else part, add a comment that properly explains that casting is legal in this specific case

Other people might even say that the < 1.9 case should remain a copy, idk.

**Comment by est31** [comment]
Other than that, looks good.

**Comment by nerzhul** [comment]
@est31 builders uses 1.8 and it's working. I think condition should be 1.7

**Comment by est31** [comment]
@nerzhul No, it should check for 1.8 too because the API didn't change. It changed between 1.8 and 1.9.

**Comment by nerzhul** [comment]
@est31 then there is no problem because jenkins & travis uses irrlicht 1.8

**Pull request closed without merging**